sycl/test/basic_tests/buffer/buffer.cpp

```diff
@@ -1,5 +1,3 @@
-// RUN: %clangxx %s -o %t1.out -lsycl -I %sycl_include
```

Comment:
Did we prohibit use of the compile to build C++ code? Otherwise we do not need to remove this code.

Reply:
This command compiles a test with static C++ RT on Windows, which causes an assert in stl.hpp
(the assert is added in this PR).

Reply:
Vladimir, Do you have good ideas on how to allow such compilation on Linux (where default C++ RT is dynamic) and not allow on Windows without -fsycl (where default C++ RT is static)?

Reply:
We have OS specific placehoders for OpenCL CPU: https://github.com/intel/llvm/blob/sycl/sycl/test/lit.cfg.py#L144
You can have some general LINUX placeholder to run the compilation only on Linux.

But it is not perfect solution so I will not insist on implementing it.

Reply:
I did not find any good way to detect if the compilation runs with -fsyntax-only or without it.